sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala

```diff
@@ -31,6 +43,21 @@ object MetalsPlugin extends AutoPlugin {
           "-P:semanticdb:failures:warning",
           s"-P:semanticdb:sourceroot:${(ThisBuild / baseDirectory).value}"
         )
+    },
+    javaSemanticdbEnabled := true,
+    javacOptions ++= {
+      if (bspEnabled.value && javaSemanticdbEnabled.value)
+        javaSemanticdbOptions.value
+      else
+        Nil
+    },
+    allDependencies ++= {
+      if (bspEnabled.value && javaSemanticdbEnabled.value)
+        List(
+          "com.sourcegraph" % "semanticdb-javac" % BuildInfo.javaSemanticdbVersion
```

Comment:
This should be added to `libraryDependencies` instead of `allDependencies`.

Also, we probably don't want this library to end up in the runtime class path of the application. So we should not put it in the `Compile` scope.

I don't know if the `CompilerPlugin` scope works with javac, probably not but you can try it with:
```scala
"com.sourcegraph" % "semanticdb-javac" % BuildInfo.javaSemanticdbVersion % Configurations.CompilerPlugin
```

Otherwise there is a notion of `CompileInternal` in sbt that should work:
```scala
"com.sourcegraph" % "semanticdb-javac" % BuildInfo.javaSemanticdbVersion % Configurations.CompileInternal
```

Reply:
Don't we actually need the compiler plugin on the normal classpath in case of Java?

Reply:
We need it in the compile class path but we don't need it in the runtime class path.

Reply:
@adpi2 ` Configurations.CompileInternal` works! thx!